bundles/io/org.openhab.io.caldav/src/main/java/org/openhab/io/caldav/internal/CalDavLoaderImpl.java

```diff
@@ -515,13 +511,58 @@ public void addEvent(CalDavEvent calDavEvent) {
                             }
                         }
                         if (query.getFilterCategory() != null) {
+                            log.trace("processing filter category");
                             if (calDavEvent.getCategoryList() == null) {
+                                log.trace("not found event category for event {}", calDavEvent.getId());
                                 continue;
                             } else {
-                                if (!calDavEvent.getCategoryList().containsAll(query.getFilterCategory())) {
+                                log.trace("processing event category");
+                                boolean eventCategoriesMatchFilterCategories = false;
+                                switch (Boolean.toString(query.getFilterCategoryMatchesAny())) {
```

Comment:
Instead of switching on a string version of the boolean, this could be done more directly with `if (query.getFilterCategoryMatchesAny()) { ... } else { ... }`.